File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
import sys
input = sys.stdin.readline

str = input().rstrip() # 백만
word = input().rstrip() # 36

stack = []
l = len(word)

for i in range(len(str)): 
    stack.append(str[i])
    if ''.join(stack[-l:]) == word:
        for _ in range(l):
            stack.pop()

if not stack:
    print('FRULA')
else:
    print(''.join(stack))

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const S = input.shift().trim(); // ""
const target = input.shift().trim(); // ""
const len = target.length;
let stack = [];

for (let i = 0; i < S.length; i++) {
  stack.push(S[i]);

  // 마지막 문자가 같으면,
  if (S[i] == target[len - 1]) {
    const last = stack.slice(-len);
    if (last.join("") === target) {
      stack.splice(-len);
    }
  }
}
// 여기서 str이 빈 문자열이면 "FRULA"를 출력하도록 수정
if (stack.length) {
  console.log(stack.join(""));
} else {
  console.log("FRULA");
}
